upgrade_guide/README.md

```diff
@@ -88,6 +88,17 @@ boundSelect = boundSelect.setInt("k", key);
 Note that, as indicated in the previous section, the public API exposes these types as interfaces:
 if for some reason you prefer a mutable implementation, it's possible to write your own.
 
+#### Prepared statement cache
```

Comment:
To clarify, it would be good to add 'This is no longer considered an anti-pattern' somewhere.